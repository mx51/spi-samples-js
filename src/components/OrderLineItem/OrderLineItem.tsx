import React, { HTMLAttributes } from 'react';
import { Button, IconButton, ListItem, ListItemText, Typography } from '@material-ui/core';
import AddCircleOutlineIcon from '@material-ui/icons/AddCircleOutline';
import currencyFormat from '../../utils/common/intl/currencyFormatter';
import useStyles from './index.styles';

interface Props extends HTMLAttributes<HTMLLIElement> {
  label: string;
  amount: number;
  onAdd?: () => void;
  disabled: boolean;
  viewOnly: boolean;
}

function OrderLineItem({ label, amount, onAdd, disabled, viewOnly, ...props }: Props): React.ReactElement {
  const classes = useStyles();

  return (
    <ListItem {...props}>
      <ListItemText
        primary={label}
        classes={{
          primary: !viewOnly ? classes.additionalChargesEnable : classes.additionalChargesDisable,
        }}
      />

      {viewOnly && <Typography>{currencyFormat(amount / 100)}</Typography>}
      {!viewOnly &&
        (amount === 0 ? (
          <IconButton
            disabled={disabled}
            aria-label="Add Amount"
            component="span"
            className={classes.addAmountBtn}
            onClick={onAdd}
          >
            <AddCircleOutlineIcon />
          </IconButton>
        ) : (
          <Button
            variant="contained"
            size="small"
            disableElevation
            disabled={disabled}
            classes={{ root: classes.additionalChargeBtn, label: classes.additionalChargeBtnLabel }}
            onClick={onAdd}
          >
            {currencyFormat(amount / 100)}
          </Button>
        ))}
    </ListItem>
  );
}

export default OrderLineItem;
