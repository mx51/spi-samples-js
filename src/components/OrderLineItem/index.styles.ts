import { createStyles, makeStyles, Theme } from '@material-ui/core';
import { red } from '@material-ui/core/colors';
import { SECONDRARY_BACKGROUND_COLOR } from '../../definitions/constants/themeStylesConfigs';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    additionalCharges: {
      color: theme.palette.info.main,
      letterSpacing: '0.02rem',
      fontSize: '1rem',
      fontWeight: 'normal',
      lineHeight: '1.5rem',
    },
    addtionalChargeBtn: {
      color: theme.palette.primary.main,
      backgroundColor: SECONDRARY_BACKGROUND_COLOR,
      fontSize: '1.25rem',
      padding: 0,
    },
    addtionalChargeBtnLabel: {
      padding: 0,
    },
    addAmountBtn: {
      padding: 0,
      color: theme.palette.info.main,
    },
  })
);
export default useStyles;
