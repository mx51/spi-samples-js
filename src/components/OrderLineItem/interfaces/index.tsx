export interface IOrderLineItemProps {
  label: string;
  amount: number;
}
