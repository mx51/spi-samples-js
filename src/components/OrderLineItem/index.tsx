import React from 'react';
import { Button, IconButton, ListItem, ListItemText, Typography } from '@material-ui/core';
import AddCircleOutlineIcon from '@material-ui/icons/AddCircleOutline';
import currencyFormat from '../../utils/common/intl/currencyFormatter';
import useStyles from './index.styles';
import { IOrderLineItemProps } from './interfaces/index';

function OrderLineItem({ label, amount, onAdd, disabled, viewOnly }: IOrderLineItemProps): React.ReactElement {
  const classes = useStyles();

  return (
    <ListItem>
      <ListItemText
        primary={label}
        classes={{
          primary: !viewOnly ? classes.additionalChargesEnable : classes.additionalChargesDisable,
        }}
      />

      {viewOnly && <Typography>{currencyFormat(amount / 100)}</Typography>}
      {!viewOnly &&
        (amount === 0 ? (
          <IconButton
            disabled={disabled}
            aria-label="Add Amount"
            component="span"
            className={classes.addAmountBtn}
            onClick={onAdd}
          >
            <AddCircleOutlineIcon />
          </IconButton>
        ) : (
          <Button
            variant="contained"
            size="small"
            disableElevation
            disabled={disabled}
            classes={{ root: classes.additionalChargeBtn, label: classes.additionalChargeBtnLabel }}
            onClick={onAdd}
          >
            {currencyFormat(amount / 100)}
          </Button>
        ))}
    </ListItem>
  );
}

export default OrderLineItem;
