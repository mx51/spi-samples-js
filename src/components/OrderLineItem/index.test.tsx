import React from 'react';
import { render } from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';
import OrderLineItem from '.';

describe('Test <OrderLineItem />', () => {
  test('snapshot for OrderLineItem component', () => {
    const onAdd = jest.fn();

    const { container } = render(<OrderLineItem label="Tip" amount={50} onAdd={onAdd} disabled={false} />);
    expect(container).toMatchSnapshot();
  });
});
