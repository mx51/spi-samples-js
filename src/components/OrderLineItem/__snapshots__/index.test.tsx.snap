// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <OrderLineItem /> snapshot for OrderLineItem component 1`] = `
<div>
  <li
    class="MuiListItem-root MuiListItem-gutters"
  >
    <div
      class="MuiListItemText-root"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary makeStyles-additionalCharges-1 MuiTypography-body1 MuiTypography-displayBlock"
      >
        Tip
      </span>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root makeStyles-addtionalChargeBtn-2 MuiButton-contained MuiButton-containedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label makeStyles-addtionalChargeBtnLabel-3"
      >
        $0.50
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </li>
</div>
`;
